lib/logstash/filters/cipher.rb

```diff
@@ -74,53 +75,139 @@ class LogStash::Filters::Cipher < LogStash::Filters::Base
   # If you want to change it, set this parameter. If you want to disable
   # it, Set this parameter to 0
   # [source,ruby]
-  #     filter { cipher { padding => 0 }}
+  #     filter { cipher { cipher_padding => 0 }}
   config :cipher_padding, :validate => :string
 
-  # The initialization vector to use
+  # The initialization vector to use (statically hard-coded). For 
+  # a random IV see the iv_random_length property
+  #
+  # NOTE: If iv_random_length is set, it takes precedence over any value set for "iv"
   #
   # The cipher modes CBC, CFB, OFB and CTR all need an "initialization
   # vector", or short, IV. ECB mode is the only mode that does not require
   # an IV, but there is almost no legitimate use case for this mode
   # because of the fact that it does not sufficiently hide plaintext patterns.
+  #
+  # For AES algorithms set this to a 16 byte string. 
+  #  
+  # 	filter { cipher { iv => "1234567890123456" }} 
   config :iv, :validate => :string
-
+  
+  # Force an random IV to be used per encryption invocation and specify
+  # the length of the random IV that will be generated via:
+  #
+  #			 OpenSSL::Random.random_bytes(int_length)
+  #
+  # If iv_random_length is set, it takes precedence over any value set for "iv"
+  #
+  # Enabling this will force the plugin to generate a unique
+  # random IV for each encryption call. This random IV will be prepended to the 
+  # encrypted result bytes and then base64 encoded. On decryption "iv_random_length" must 
+  # also be set to utilize this feature. Random IV's are better than statically
+  # hardcoded IVs
+  #
+  # For AES algorithms you can set this to a 16
+  #  
+  # 	filter { cipher { iv_random_length => 16 }} 
```

Comment:
Generally IVs are equal to block size. For AES, this is 128bits. I think the configuration options for this are sufficient for someone knowing what they are doing to be able to tweak it to what they want should they use something other that AES. If you are asking if we should hardwire a dictionary or something of algorithm to iv block size, I'm not sure this is necessary.